### DISCUSSION TITLE
importer callback returns Unable to connect

### DISCUSSION THREAD

**Discussion by adan-ea** [firefly-iii/firefly-iii#9339] [Firefly III] [answered]
Hi there, 
First of all, I know it's a me issue and I'm sorry about this but here is my problem: 
I have firefly and the importer set with docker compose as said in the documentation. The thing is, it's on my local network on a mini pc. which means no `localhost:81` for me. Here is what I got so far: 
firefly: `http://192.168.1.25:3009/profile` (to create my Oauth token for the importer. Client ID = `2`)

For the redirect URL of the client, I tried setting `http://x/callback`
where x is : `importer:8080`, `localhost:8080`, `192.168.1.25:3010`, `importer:3010`

f-importer : `http://192.168.1.25:3010/token`
And firefly says the callback URL for my installation is `http://192.168.1.25:3010/callback`
The thing is when I use the ClientID and submit. 
I get this from firefox 
![{B52BC34A-0F41-49DE-A78D-6865885D278C}](https://github.com/user-attachments/assets/7d4d6673-bce9-42ee-af67-e57326c2b1a6)

at this URL : 
http://localhost/oauth/authorize?client_id=2&redirect_uri=http%3A%2F%2F192.168.1.25%3A3010%2Fcallback&response_type=code&scope=&state=x&code_challenge=x&code_challenge_method=S256

**Comment by adan-ea** [comment]
Update: I removed the vanity url from .importer.env (previously http://localhost)
and left FIREFLY_III_URL to FIREFLY_III_URL=http://app:8080 

and now I get a new error 
![{C348C4D0-4E4A-40E7-AE5C-2B51D10F9BAD}](https://github.com/user-attachments/assets/fee98b2e-828d-4a23-a090-177be178f3a6)
![{3AC68A79-B6EE-4FF2-A498-68069429298B}](https://github.com/user-attachments/assets/3f78d8f4-c735-492b-be30-c41ff4dc76ad)

**Comment by adan-ea** [comment, marked as answer]
here is the answer: in my case just putting the actual ip with port solved it. Without using app:8080 it worked like a charm, sorry for the notifications

**Reply by JC5** [reply]
Not a problem, glad to hear it works!
